crypto/ec/ecp_sm2p256.c

```diff
@@ -0,0 +1,790 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ */
+
+/*
+ * SM2 low level APIs are deprecated for public use, but still ok for
+ * internal use.
+ */
+#include "internal/deprecated.h"
+
+#include <string.h>
+#include <openssl/err.h>
+#include "crypto/bn.h"
+#include "ec_local.h"
+#include "internal/constant_time.h"
+
+#if defined(__GNUC__)
+# define ALIGN32 __attribute((aligned(32)))
+# define ALIGN64 __attribute((aligned(64)))
```

Comment:
I noticed ALIGN32 are declared in two .c files. one is here. the other is in [L41 of crypto/ec/ecp_nistz256.c](https://github.com/openssl/openssl/blob/c612289b77c37f7295d5af0d0e6b6c04e6ba727c/crypto/ec/ecp_nistz256.c#L41)

Considering this as a common feature, should we put these macros into a common file? such as ec_local.h in crypto/ec/?

Reply:
They would be better in a header file.  `include/internal/e_os.h` seems like a reasonable place.

Reply:
Should this change be made as part of this pull request, or can it be done in another, later, one?

Reply:
I think doing it later would be OK, if someone really does it :grin:

Reply:
I _can_ do that!

So can we get approvals on this PR and get it merged?